libs/community/langchain_community/llms/bedrock.py

```diff
@@ -353,7 +360,16 @@ def _identifying_params(self) -> Mapping[str, Any]:
         }
 
     def _get_provider(self) -> str:
-        return self.model_id.split(".")[0]
+        if self.provider:
+            return self.provider
+        else:
+            if self.model_id.startswith("arn"):
+                raise ValueError(
+                    "Model provider should be supplied when passing a model ARN as "
+                    "model_id"
+                )
+            else:
+                return self.model_id.split(".")[0]
```

Comment:
else is extraneous here.

```suggestion
        if self.model_id.startswith("arn"):
            raise ValueError(
                "Model provider should be supplied when passing a model ARN as "
                "model_id"
            )
        else:
            return self.model_id.split(".")[0]
```